libs/datasets/timeseries.py

```diff
@@ -216,7 +216,7 @@ def __post_init__(self):
 
     @property
     def date_indexed(self) -> pd.DataFrame:
-        return self.data.set_index(CommonFields.DATE)
+        return self.data.set_index(CommonFields.DATE).asfreq("D")
```

Comment:
in some cases (as was the case in `test_calculate_test_positivity`) the index was a date index, but did not have a frequency, causing comparisons against otherwise-equal dataframes/series to fail. 

This explicitly adds the "day" frequency to the date index